### ISSUE TITLE
Spies are OP

### ISSUE THREAD

**Issue by AMHOL** [OpenRA/OpenRA#12228]
TLDR:

Spies are OP, I think there needs to be some discussion around how best to nerf them

Spies can be so hard to spot, I was playing a game earlier with a player I suspect was Christian, (Drunk) SoScared and Kazu, as well as two other players, but this also applies in a 1v1. Kazu sent about 20 spies into Christian's base, Christian was being attacked from two fronts and had 45 APM in an hour long game, so he didn't notice, he then continually sent spies into Christian's power plants, low power is such a set-back, also, this was an FFA on a large map, so Christian was floating around 30k, had Kazu sent the spies into the refinery he could have literally stolen the entirety of Christian's cash with a single click, it seems fair that a player can be punished for floating so much cash, but all players were floating in this game, and such severe punishment seems unjust.

When you're concentrating on micro managing your units spotting a spy can be so difficult, this applies to a game of any size, whether team or 1v1 IMO.

There is also the possibility of "spy locking", chrono over an MCV, put a spy on hold at 1 second with 6 barracks, then build the 7th at your enemys base and spy the power, refinery, or pretty much anything you like, this is practically impossible to stop, because once you are in low power, or 30k has been reduced to 3,750 by 3 spies, you can't build any units to counter. I have watched this happen on several occasions, and experienced it myself on at least one memorable occasion on Sidestep, I had control of `literally every expansion`, I had a 15+ APM lead on the player I was playing against (and I don't waste actions) and a spy lock lost me the game.

**Comment by keklylel** [comment]
I agree, spies can be very overpowered if used correctly. However spies have a counter, guard dogs.

**Comment by StexoO** [comment]
See https://github.com/OpenRA/OpenRA/issues/9110 about low power.

**Comment by AMHOL** [comment]
@StexoO thanks, not keen on that solution, I was thinking adding a per-building cooldown timer on infiltration could be good. I.e. if a spy infiltrates a refinery that refinery can't be infiltrated again for 30(n) seconds, that way you can't just spam spies into a ref and steal the entire bank, and power down requires some skill as you will have to infiltrate different power plants to keep them in low power. This would also fix the spy lock when base pushing. Be good to get some opinions on this.

**Comment by matjaeck** [comment]
@AMHOL still concerned about imbalanced spies now that they can not steal money anymore? (I also hate them).

:+1: for the infiltration cooldown per structure but this should probably be discussed in its own ticket ("x is OP" tickets are usually not very productive). Unfortunately #15281 died because this would be needed for this I think.

**Comment by matjaeck** [comment]
IMO it would be more constructive to have a separate issue for this feature request (cooldown for entering certain buildings) but that does necessarily mean that this issue needs to be closed.

**Comment by Smittytron** [comment]
Related: #12791

**Comment by abcdefg30** [comment]
Closing as stale.
